### PR TITLE
WIP: Update example to use spec object and units

### PR DESCRIPTION
**Pull request by samaloney**
## PR Description

Update exiting fitting demo to use units and work with astropy modelling.

### CODE REVIEW

**Review by settwi — COMMENTED** [review]
Main thing that i would like to see changed is removal of the comment blocks

This is also a big PR. could you add a bit to the conversation about what tests are in place that correspond to the changes? it's a bit hard to parse through everything having not worked on it

finally i think we should revisit the bin centers vs edges question. i think having the option to do both is confusing and could lead to strange errors for folks doing analysis down the road

File: sunkit_spex/spectrum/spectrum.py [hunk 1]
```diff
@@ -85,6 +85,8 @@ def __new__(cls, value, *args, bin_specification="centers", **kwargs):
 
         if bin_specification == "edges":
             obj._bin_edges = bin_edges
+        elif bin_specification == "centers":
+            obj._bin_edges = None
```
[the remark this answers is not in the document]
Comment (jajmitchell) [reply sunkit_spex/spectrum/spectrum.py:89]: Agreed!

**Comment by jajmitchell** [comment]
@settwi I will take a proper look through your comments and make some changes later in the week, there are some things that definitely need changing and I will add some more description of all the changes that I have made. I am giving the workshop tomorrow and so need this branch to remain the same until after tomorrow

**Comment by settwi** [comment]
@jajmitchell what is the status of this branch?

**Comment by jajmitchell** [comment]
@settwi I think that its best that this PR is broken down into a few smaller PRs with more defined scope, particularly as much will need to be changed, due to the decisions made regarding the spectrum object and fitter etc
